staging/src/k8s.io/kubectl/pkg/util/service_port.go

```diff
@@ -28,13 +28,13 @@ import (
 // It returns an error when a named port can't find a match (with -1 returned), or when the service does not
 // declare such port (with the input port number returned).
 func LookupContainerPortNumberByServicePort(svc v1.Service, pod v1.Pod, port int32) (int32, error) {
+	if svc.Spec.ClusterIP == v1.ClusterIPNone {
+		return port, nil
```

Comment:
I'm not familiar with this code, but if the service is headless, why the same rules doesn't apply for ports @thockin ?

Reply:
which ports do you mean, target ports?

Reply:
First let me see if I understand this correctly :)
So, you specify a `kubectl port-forward service/myservice 8443` and you try to find the container port that belongs to the service port 8443, right?

is mostly curiosity, I see that it's also mentioned in the function description
> as well as ignoring targetPort when clusterIP=None

but why the headless services doesn't look for the Spec.Port and  Spec.TargetPort too?

Reply:
ok, scratch my comment
https://github.com/kubernetes/kubernetes/blob/ddf3eb5a1877338da806c2be15b53f456b6342c9/pkg/apis/core/types.go#L3750-L3752

Reply:
right, as I understand it (may be wrong), headless service means `ClusterIP` is none and `TargetPort`  = `Port` of pod

Reply:
I don´t feel this is correct, because this will forward a port despite the service doesn´t have any port defined 

Create a headless service without a port:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: headless-service
spec:
  clusterIP: None
  selector:
    app: api
```
Create a pod for the service
```
kubectl run test --image httpd --port 80 -l app=api
```

when trying to forward a port it fails, because the service doesn´t have any port defined:
```
kubectl port-forward service/headless-service 8000 
error: Service headless-service does not have a service port 8000
```
Current behaviour checks that the Port that we are asking to forward matches the Service Port, with this PR you are removing this check.

Reply:
Thanks so  much for your reviewing, i'll close this PR and keep on studying and reading the source codes